### ISSUE TITLE
SQL error when doing batch search

### ISSUE THREAD

**Issue by at055612** [gchq/stroom#3043]
open a dashboard, press the process button, set the timeout options, press OK, BOOM

Seen in v7.0.1

```
org.jooq.exception.DataAccessException: SQL [insert into `processor_filter` (`create_time_ms`, `create_user`, `update_time_ms`, `update_user`, `uuid`, `fk_processor_id`, `fk_processor_filter_tracker_id`, `priority`, `reprocess`, `enabled`, `deleted`, `min_meta_create_time_ms`, `max_meta_create_time_ms`, `version`) values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)]; Field 'data' doesn't have a default value
```

**Comment by at055612** [comment]
Issue is failure to marshal an empty str in `QueryData.params`.  Need to set it to null or a non-empty str

**Comment by at055612** [comment]
Fixed in >7.0.1
